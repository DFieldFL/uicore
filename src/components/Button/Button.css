.button {
  &[class*='bp3-button'],
  &[class*='bp3-button']:not([class*='bp3-intent-']) {
    /* Override Blueprint styles */
    border-radius: 4px;
    display: inline-flex;
    font-family: var(--font-family);
    font-size: var(--font-size-normal);
    height: 32px;
    padding: 5px 10px;
    box-shadow: none;
    background-image: none;
    transition: all 0.25s cubic-bezier(0.39, 0.58, 0.57, 1);

    /* No intent/None intent */
    border: 1px solid var(--grey-350);
    color: var(--grey-600);
    background-color: var(--white);

    &.link {
      padding: 0;
      border: none;
      height: auto;
      min-height: auto;
      color: var(--blue-500);
      background-color: transparent;
      vertical-align: unset;

      &:hover,
      &:active,
      &[class*='active'] {
        background: none;
        border: none;
        box-shadow: none;
      }

      &:hover:not([disabled]) {
        text-decoration: underline;
        color: var(--blue-600);
      }
    }

    &:hover {
      border: 1px solid var(--blue-400);
      color: var(--blue-500);
      background-color: var(--blue-100);
      box-shadow: none;
    }

    &:active {
      border: 1px solid var(--blue-400);
      color: var(--blue-500);
      background-color: var(--blue-200);
    }

    &[disabled] {
      border: 1px solid var(--grey-350);
      color: var(--grey-600);
      background-color: var(--white);
      opacity: 0.5;

      &.link {
        background-color: none;
        border: none;
      }
    }

    &[class*='minimal'] {
      border: none;
      background: none;

      /* &:hover {
        background-color: var(--blue-100);
      }

      &:active {
        background-color: var(--blue-200);
      } */
    }

    /* Primary */
    &[class*='intent-primary'] {
      border: 1px solid var(--blue-500);
      color: var(--white);
      background-color: var(--blue-500);

      &:hover {
        border: 1px solid var(--blue-600);
        background-color: var(--blue-600);
      }

      &:active {
        border: 1px solid var(--blue-800);
        background-color: var(--blue-800);
      }

      &[disabled] {
        border: 1px solid var(--blue-500);
        background-color: var(--blue-500);
      }

      &[class*='minimal'] {
        border: none;
        background: none;
        color: var(--blue-500);

        &:hover {
          background-color: var(--blue-100);
        }

        &:active {
          background-color: var(--blue-200);
        }
      }
    }

    /* Success */
    &[class*='intent-success'] {
      color: var(--white);
      border: 1px solid var(--green-500);
      background-color: var(--green-500);

      &:hover {
        border: 1px solid var(--green-600);
        background-color: var(--green-600);
      }

      &:active {
        border: 1px solid var(--green-800);
        background-color: var(--green-800);
      }

      &[disabled] {
        border: 1px solid var(--green-500);
        background-color: var(--green-500);
      }
    }

    /* Warning */
    &[class*='intent-warning'] {
      color: var(--white);
      border: 1px solid var(--yellow-500);
      background-color: var(--yellow-500);

      &:hover {
        border: 1px solid var(--yellow-600);
        background-color: var(--yellow-600);
      }

      &:active {
        border: 1px solid var(--yellow-800);
        background-color: var(--yellow-800);
      }

      &[disabled] {
        border: 1px solid var(--yellow-500);
        background-color: var(--yellow-500);
      }
    }

    /* Danger */
    &[class*='intent-danger'] {
      color: var(--white);
      border: 1px solid var(--red-500);
      background-color: var(--red-500);

      &:hover {
        border: 1px solid var(--red-600);
        background-color: var(--red-600);
      }

      &:active {
        border: 1px solid var(--red-800);
        background-color: var(--red-800);
      }

      &[disabled] {
        border: 1px solid var(--red-500);
        background-color: var(--red-500);
      }
    }
  }

  :global {
    .bp3-icon {
      color: currentColor !important;
      margin-right: 5px;
    }

    .bp3-spinner .bp3-spinner-head {
      stroke: currentColor;
    }

    .bp3-button-text {
      margin-right: 5px;
    }
  }

  /* Make inline icons have the same color as the button color */
  > span {
    svg {
      width: 12px;
      height: 12px;

      path,
      circle {
        fill: currentColor !important;
      }

      :not(path) {
        stroke: currentColor !important;
      }
    }

    &:only-child svg {
      width: 16px;
      height: 16px;
    }
  }

  /* Disabled */
  /* stylelint-disable no-descending-specificity */
  &[class*='bp3-disabled'] {
    svg,
    svg > * {
      cursor: not-allowed;
    }
  }
}
