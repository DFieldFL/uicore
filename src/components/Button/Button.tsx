import React from 'react'
import { Button as BButton, IButtonProps, Classes } from '@blueprintjs/core'
import css from './Button.css'

export interface ButtonProps extends Omit<IButtonProps, 'icon'> {
  icon?: IButtonProps['icon'] | React.FunctionComponent<any>
  children?: React.ReactChild
}

function Button(props: ButtonProps) {
  let { icon } = props

  if (typeof icon === 'function') {
    const Icon = icon as React.ElementType
    icon = (
      <span className={Classes.ICON}>
        <Icon />
      </span>
    )
  }

  return <BButton {...props} icon={icon} className={css.button} />
}

export { Button }
