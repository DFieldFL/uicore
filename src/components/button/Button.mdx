---
name: Button
---

import Button from './Button'

# Hello world

<Button id="sample-button" text="Platform Button" icon="add" />
