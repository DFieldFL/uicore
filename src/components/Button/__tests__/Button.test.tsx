import React from 'react'
import { render } from '@testing-library/react'

import { Button } from '../Button'

describe('<Button /> tests', () => {
  test('snapshot test', () => {
    const { container } = render(<Button>Click Me!</Button>)

    expect(container).toMatchSnapshot()
  })

  test('renders icon as functional components', () => {
    const { container } = render(<Button icon={() => <div>ICON!</div>}>Click Me!</Button>)
    expect(container).toMatchSnapshot()
  })
})
