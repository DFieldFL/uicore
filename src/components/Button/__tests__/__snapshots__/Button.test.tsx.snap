// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests renders icon as functional components 1`] = `
<div>
  <button
    class="bp3-button button"
    type="button"
  >
    <span
      class="bp3-icon"
    >
      <div>
        ICON!
      </div>
    </span>
    <span
      class="bp3-button-text"
    >
      Click Me!
    </span>
  </button>
</div>
`;

exports[`<Button /> tests snapshot test 1`] = `
<div>
  <button
    class="bp3-button button"
    type="button"
  >
    <span
      class="bp3-button-text"
    >
      Click Me!
    </span>
  </button>
</div>
`;
